test/jdk/java/net/httpclient/http2/BadHeadersTest.java

```diff
@@ -127,23 +131,23 @@ public class BadHeadersTest {
                 return frames;
             };
 
-    @DataProvider(name = "variants")
-    public Object[][] variants() {
-        return new Object[][] {
-                { http2URI,  false, oneContinuation },
-                { https2URI, false, oneContinuation },
-                { http2URI,  true,  oneContinuation },
-                { https2URI, true,  oneContinuation },
-
-                { http2URI,  false, byteAtATime },
-                { https2URI, false, byteAtATime },
-                { http2URI,  true,  byteAtATime },
-                { https2URI, true,  byteAtATime },
-        };
+    static Stream<Arguments> variants() {
+        return Stream.of(
+                Arguments.of(http2URI,  false, oneContinuation),
+                Arguments.of(https2URI, false, oneContinuation),
+                Arguments.of(http2URI,  true,  oneContinuation),
+                Arguments.of(https2URI, true,  oneContinuation),
+
+                Arguments.of(http2URI,  false, byteAtATime),
+                Arguments.of(https2URI, false, byteAtATime),
+                Arguments.of(http2URI,  true,  byteAtATime),
+                Arguments.of(https2URI, true,  byteAtATime)
+        );
```

Comment:
AFAICT, you don't need to return a `Stream<Arguments>`, the old `Object[][]` should suffice. Maybe that is better to revert this to old style to keep the diff minimal – granted `Object[][]` indeed works with JUnit.

Reply:
If this was a new tests then IMO using `Arguments.of(...)` would be preferable. Since this is an old test then arguably keeping the old Object[][] could make backports easier.